src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1108,6 +1108,322 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
+
+      // If the DC was reloaded the internal q_field_map was not restored
+      // so we can't just use HasQField here. We should fix this in the  Load
+      // function. 
+      if (basis_name.substr(0, basis_name.find(delimiter)) != "QF")
+      {
+        SLIC_ERROR("field name provided was not for a QF");
+        return -1;
+      }
+      // We know our basis string should be something like:
+      // QF_Default_[ORDER]_[VDIM]
+      pos = basis_name.find(delimiter);
+      basis_name.erase(0, pos + delimiter.length());
+
+      pos = basis_name.find(delimiter);
+      basis_name.erase(0, pos + delimiter.length());
+      std::string order = basis_name.substr(0, basis_name.find(delimiter));
+
+      return std::stoi(order);
+   }
+   else
+   {
+      SLIC_WARNING("field " << field_name << " does not exist");
+      return -1;
+   }
+
+}
+
+int MFEMSidreDataCollection::GetQFieldVDim(const std::string &field_name)
+{
+   sidre::Group* grp = nullptr;
+   if( m_bp_grp->hasGroup("fields/" + field_name))
+   {
+      grp = m_bp_grp->getGroup("fields/" + field_name);
+
+      sidre::View *v = grp->getView("basis");
+      std::string basis_name = v->getString();
+
+      size_t pos = 0;
+      std::string delimiter = "_";
+
+      // If the DC was reloaded the internal q_field_map was not restored
+      // so we can't just use HasQField here. We should fix this in the  Load
+      // function. 
+      if (basis_name.substr(0, basis_name.find(delimiter)) != "QF")
+      {
+        SLIC_ERROR("field name provided was not for a QF");
+        return -1;
+      }
+
+      // We know our basis string should be something like:
+      // QF_Default_[ORDER]_[VDIM]
+      pos = basis_name.find(delimiter);
+      basis_name.erase(0, pos + delimiter.length());
+
+      pos = basis_name.find(delimiter);
+      basis_name.erase(0, pos + delimiter.length());
+
+      pos = basis_name.find(delimiter);
+      basis_name.erase(0, pos + delimiter.length());
+
+      return std::stoi(basis_name);
+   }
+   else
+   {
+      SLIC_ERROR("field " << field_name << " does not exist");
+      return -1;
+   }
+}
+
+double* MFEMSidreDataCollection::GetQFieldData(const std::string &field_name)
+{
+   sidre::Group* grp = nullptr;
+   if( m_bp_grp->hasGroup("fields/" + field_name))
+   {
+      grp = m_bp_grp->getGroup("fields/" + field_name);
+
+      // If the DC was reloaded the internal q_field_map was not restored
+      // so we can't just use HasQField here. We should fix this in the  Load
+      // function. 
+      sidre::View *v = grp->getView("basis");
+      std::string basis_name = v->getString();
+
+      std::string delimiter = "_";
+
+      if (basis_name.substr(0, basis_name.find(delimiter)) != "QF")
+      {
+        SLIC_ERROR("field name provided was not for a QF");
+        return nullptr;
+      }
+
+      v = grp->getView("values");
+      void* vptr = v->getVoidPtr();
+      return static_cast<double*>(vptr) + v->getOffset();
```

Comment:
Minor:  This might be a little clearer (and also maybe more type-safe, I think the access through `conduit::Value` is type-checked):
```suggestion
      return v->getData();
```